sample/build.gradle

```diff
@@ -59,6 +53,14 @@ android {
   }
 }
 
+configurations.configureEach {
+  resolutionStrategy.eachDependency {
+    if (requested.group == "androidx.lifecycle" && requested.name == "lifecycle-viewmodel") {
+      useVersion versionFor(project, AndroidX.lifecycle.viewModel)
+    }
```

Comment:
AFAICS release artifacts are never built (which means it is not tested if library obfuscates correctly?). If you try to run `assembleRelease` you'll get `Duplicate class androidx.lifecycle.ViewModelLazy found in modules (...)` error. 

I'm proposing to enforce single artifact version across all configurations

Reply:
Is there an incompatibility between two androidx versions that we could solve by using the right dependencies?

Reply:
It's not really _incompatibility_ of libraries, but a result of the build setup. The build (transitively) pulls both
- `androidx.lifecycle:lifecycle-viewmodel:2.5.1` (by `androidx.appcompat:appcompat`)  
and 
- `androidx.lifecycle:lifecycle-viewmodel-ktx:2.2.0` (by i.e. `androidx.paging:paging-runtime-ktx`). 

It works in debug variant because `debugImplementation libs.androidx.fragment.testing` depends on viewmodel-ktx in _proper_ version, but in release mode there is no such dependency 🤷 

I pushed an update, that adds the `viewmodel-ktx` dependency regardless of the build variant (so it's not a transitive dependency anymore)